src/diffpy/labpdfproc/tests/test_tools.py

```diff
@@ -4,7 +4,80 @@
 import pytest
 
 from diffpy.labpdfproc.labpdfprocapp import get_args
-from diffpy.labpdfproc.tools import known_sources, load_user_metadata, set_output_directory, set_wavelength
+from diffpy.labpdfproc.tools import (
+    known_sources,
+    load_user_metadata,
+    set_input_files,
+    set_output_directory,
+    set_wavelength,
+)
+from diffpy.utils.parsers.loaddata import loadData
+
+# Use cases can be found here: https://github.com/diffpy/diffpy.labpdfproc/issues/48
+params_input = [
+    (["good_data.chi"], [".", "good_data.chi"]),
+    (["input_dir/good_data.chi"], ["input_dir", "good_data.chi"]),
+    (["./input_dir/good_data.chi"], ["input_dir", "good_data.chi"]),
+    (
+        ["."],
+        [
+            ".",
+            ["good_data.chi", "good_data.xy", "good_data.txt", "unreadable_file.txt", "binary.pkl"],
+        ],
+    ),
+    (
+        ["./input_dir"],
+        [
+            "input_dir",
+            ["good_data.chi", "good_data.xy", "good_data.txt", "unreadable_file.txt", "binary.pkl"],
+        ],
+    ),
+    (
+        ["input_dir"],
+        [
+            "input_dir",
+            ["good_data.chi", "good_data.xy", "good_data.txt", "unreadable_file.txt", "binary.pkl"],
+        ],
+    ),
+]
+
+
+@pytest.mark.parametrize("inputs, expected", params_input)
+def test_set_input_files(inputs, expected, user_filesystem):
+    expected_input_directory = Path(user_filesystem) / expected[0]
+    expected_input_files = expected[1]
+
+    cli_inputs = ["2.5"] + inputs
+    actual_args = get_args(cli_inputs)
+    actual_args = set_input_files(actual_args)
+    assert actual_args.input_directory == expected_input_directory
+    assert set(actual_args.input_file) == set(expected_input_files)
+
+
+params_input_bad = [
+    (["new_file.xy"]),
+    (["./input_dir/new_file.xy"]),
+    (["./new_dir"]),
+]
+
+
+@pytest.mark.parametrize("inputs", params_input_bad)
+def test_set_input_files_bad(inputs, user_filesystem):
+    cli_inputs = ["2.5"] + inputs
+    actual_args = get_args(cli_inputs)
+    with pytest.raises(ValueError):
+        actual_args = set_input_files(actual_args)
+
+
+def test_loadData_with_input_files(user_filesystem):
```

Comment:
This seems to be testing your test, not anything in the code?  I guess this is ok, but maybe make a comment to that effect?